### DISCUSSION TITLE
Notification on remote start generator and transfer?

### DISCUSSION THREAD

**Discussion by Nate52441** [jgyates/genmon#1211] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).
- [X] I have included the generator make and model in my submission.

### Version

1.19.03/generac power pact

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have submitted logs

### What is/are your question(s)?

Finally got everything setup on my generator and I’m confirming everything is working as expected from genmon in regards to monitoring. I setup email notifications and confirmed it’s working, however when I use “start generator and transfer” from genmon it doesn’t send me any notification. Are notifications suppressed when trigger start/stop from genmon? I can see the info in the logs and have all notification types enabled for my email.  I couldn’t find anything relevant to the topic in the docs. I know I’ve submitted a few questions the past few days so I greatly appreciate all the assistance.

**Comment by jgyates** [comment]
@Nate52441 

A few questions / comments:

1) you said you are using version 1.19.03. The latest is 1.19.04. Your version is at least 6 months out of date
2) Notifications are sent when the generator state changes. When you issue the remote start and transfer command does you generator start and activate the transfer switch? If it does not then there may be another issue besides notifications causing the problem. If an alarm is active this could cause the remote commands not to work. Does the remote start (no transfer) command work? The power pact has limited functionality of a standard Evolution controller so validating the remote commands may be the first thing to try.
3) You mentioned that you sent logs. I have not received any logs. You outbound email has to be setup and functioning to send log from the About page.

**Reply by Nate52441** [reply]
1. I’m working on updating, it failed when I tried to do it earlier
2. The generator does start and transfers as expected and when issue the stop command it’s stops and cools down 
3. I have email setup and working, I received the test email when I set it up. I’ll send them again.

**Reply by Nate52441** [reply]
I tested again and got some notifications but they’re unintelligible (if they’re meant to look that way I apologies)

here are some examples

FRM:Genmon 
SUBJ:Generator Notice at SiteName
MSG:
Time: 03:33:58 PM
Date: Wed 13-Nov-2024
Generator Rea

FRM:Genmon 
SUBJ:Generator Notice at Spring hill generato
MSG:
Time: 03:38:10 PM
Date: Wed 13-Nov-2024
System Health : System Initializing. 


for whatever reason I keep getting the system initializing notification multiple times.

**Reply by jgyates** [reply]
I received your logs.

FYI, if you did not use the git command to install genmon as described [here](https://github.com/jgyates/genmon/wiki/3.3--Setup-genmon-software) then software updates will not work.

Your logs indicate that you have a marginal connection. Since you are using TCP for the link and you are seeing timeout errors I am assuming that you are using Wifi for your network connection. This is likely the cause of your problem. A poor wifi signal will cause communication errors with the generator and errors when sending email (if you are sending email of wifi as well).

You are getting many of these errors in your log file when sending email:

2024-11-13 15:33:12,024 : Error SMTP sendmail: Connection unexpectedly closed : mymail.py:699
2024-11-13 15:33:12,025 : Error in SendMailThread, sendEmailDirectMIME failed, retrying

This error is when the software can no longer communicate with the email server when trying to send and email.

You are getting the system initializing multiple times due to the bad wifi signal. Genmon can not communicate over your wifi network as the signal is to low. 

Also, the messages you are showing above look like they are going through an email to text message service. Are you using the add on to send text messages vai email or are you expecting and email? The content of the email is very different vs the short message notification. 

Looks like you have two issues, both are caused by bad wifi.

1) genmon can not communicate to the generator without errors using modbus over TCP
2) network communications needed for notifications are not occurring without errors

Using Ethernet instead of wifi or using serial instead of TCP would correct the issue

**Comment by Nate52441** [comment]
It’s weird that the network errors are happening on the software side, it’s
a docker container running on my laptop since im waiting for my PI to
arrive. Hopefully migration to PI will resolve that.

As far as WiFi goes these are pretty old ubiquiti APs so it’s probably time
for an upgrade anywhow. I’ll try to resolve those issues and report back.

**Reply by jgyates** [reply]
Maybe try turning off your wifi on the laptop and just plug it in via Ethernet? The the only wireless connection would be from your serial to TCP adapter (assuming it is wifi based an not Ethernet based)

**Reply by Nate52441** [reply]
Couldn’t hurt to plug in of course, I’ll give that a go. Thanks again for all the help, I sent a donation your way as a thank you.
